ui/src/app/utils/access-utils.tsx

```diff
@@ -402,9 +403,14 @@ export const useNeedsToAcceptTOS = () => {
       // before checking term of service status
       (async () => {
         try {
-          const userHasAcceptedLatestTOS =
-            await profileApi().getUserTermsOfServiceStatus();
-          setUserRequiredToAcceptTOS(!userHasAcceptedLatestTOS);
+          // Do not show terms of service page, if the user is disabled or not eligible for RT
+          if (!userIsEnabled(profile.disabled)) {
```

Comment:
I could also move this check to app-routing L228. :
const doesUserNeedToAcceptTOS = useNeedsToAcceptTOS() && userIsEnabled(profile.disabled) ;

Reply:
I would prefer to keep all the logic for "should we show TOS" here, but you could convince me that app-routing is the right place.

I also think we can simplify this a good deal.  What do you think of this?

* Rename this function `shouldShowTOS()` or similar
* Remove `useState() and useEffect()`
* Change the `setUserRequiredToAcceptTOS()` calls to `return`s

Reply:
No, now that i looking at the code again, it makes sense to keep it at one place rather than add AND conditions

As for simplifying the logic,  i think we used useEffect to wait for profile object to be populated. But that was a long time ago ,i'll check again